src/components/TodoComponent/TodoComponent.tsx

```diff
@@ -0,0 +1,117 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  isLoading: boolean,
+  onDelete: (id: number) => void
+  onUpdateTodo: (id: number, todo: object) => void
+};
+
+export const TodoComponent: React.FC<Props> = ({
+  todo,
+  isLoading,
+  onDelete,
+  onUpdateTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(todo.title);
+
+  const handleRenameSubmition = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      onDelete(todo.id);
+    } else {
+      onUpdateTodo(todo.id, { title: todoTitle });
```

Comment:
Instead of using an inline object, create a new object with the new title and pass it to `onUpdateTodo` function. This way, it's easier to understand the changes made to the todo object.